src/sample.py

```diff
@@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
 		return True
 debugger = Debugger()
 hb.buffer_set_message_func (buf, debugger.message, 1, 0)
-hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
+hb.buffer_add_codepoints (buf, codepoints, 0, -1)
```

Comment:
so it does the exact same thing except not validate? is it worth pointing
that out in the/any documentation or something since 99.9% of the time
you’re not going to get invalid strings in python at least

On Fri, Jun 17, 2016 at 7:59 PM, Khaled Hosny <notifications@github.com>
wrote:

> In src/sample.py
> <https://github.com/behdad/harfbuzz/pull/269#discussion_r67589679>:
>
> > @@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
> >  		return True
> >  debugger = Debugger()
> >  hb.buffer_set_message_func (buf, debugger.message, 1, 0)
> > -hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
> > +hb.buffer_add_codepoints (buf, codepoints, 0, -1)
>
> See bcba8b4
> <https://github.com/behdad/harfbuzz/commit/bcba8b45024e1eca8be77ca2657de1dc44dbf8fb>
> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/behdad/harfbuzz/pull/269/files/0a9c03bc45e82c95377431576ddc92e8cd32ddb4#r67589679>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACcEOtYnYMrKZbfRGkXRkioHLPRkzfByks5qMzTkgaJpZM4I3_8R>
> .
>